cmd/fluxd/main.go

```diff
@@ -186,6 +192,16 @@ func main() {
 		*sshKeygenDir = *k8sSecretVolumeMountPath
 	}
 
+	// Import GPG keys, if we've been told where to look for them
+	if *gitImportGPG != "" {
+		keyfiles, err := gpg.ImportKeys(*gitImportGPG)
+		if err != nil {
+			logger.Log("error", "failed to import GPG keys", "err", err.Error())
```

Comment:
Shouldn't we refuse to start if the import doesn't succeed?

Reply:
Only write operations to git will fail if the import does not succeed (but only if `gitSigningKey` is set), all other parts of Flux will continue to work.

Refusing to start because Flux _may_ not be able to do a part of its job is a bit extreme imho.

Reply:
My concern is that the import failure can go undetected unless we fail to start. The user can choose to avoid the import as a response.

No super strong opinion though

Reply:
I think due to expectations of the user about what Flux does this will not go undetected as it halts the rollout of new images (and will throw an error in the logs every time Flux attempts to update a manifest).